pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -473,6 +481,225 @@ func (p *Provider) GetVMSpecs(
 	return vmSpecs, nil
 }
 
+func (p *Provider) GetVMArchitecture(l *logger.Logger, v *vm.VM) (vm.CPUArch, error) {
+
+	// If VM is tagged with arch FIPS, we use it
+	archLabel, ok := v.Labels["arch"]
+	if ok && archLabel == "fips" {
+		return vm.ArchFIPS, nil
+	}
+
+	// Otherwise, we query the CPU architecture from the machine type
+	return p.getCPUArchFromMachineType(v.MachineType, v.Zone, WideCacheKey)
+}
+
+func (p *Provider) getCPUArchFromMachineType(
+	machineType, zone string, ckl CacheKeyLookup,
+) (vm.CPUArch, error) {
+
+	// Note: zone is specified as an empty string as machine type architectures are the same
+	// across all zones and we want to have as many cache hits as possible
+	metadata, err := p.getMachineTypeSpecs(machineType, p.GetProject(), zone, ckl)
+	if err != nil {
+		return vm.ArchUnknown, err
+	}
+
+	// If arch is unspecified, GCP's default is amd64
```

Comment:
Is this default documented somewhere? Let's include the link; otherwise, let's clarify that it's an assumption based on observations.

Reply:
Closest to a documentation is [here](https://cloud.google.com/compute/docs/images/create-custom#considerations_for_arm_images):
```
To help your users differentiate between Arm and x86 images, Arm images will have an architecture field set to ARM64. Possible values for this field are:

ARCHITECTURE_UNSPECIFIED
X86_64
ARM64
```

This hints that this field is only set on Arm images while not clearly state that it's not present on x86 images (which is my observation).